### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by yuokada** [yuokada/embulk-output-orc#7]
```
2017-10-15 19:01:18.890 +0900 [WARN] (embulk-output-executor-3): Owner thread expected Thread[0014:task-0000,5,main], got Thread[embulk-output-executor-3,5,main]
2017-10-15 19:01:18.891 +0900 [WARN] (embulk-output-executor-0): Owner thread expected Thread[0014:task-0000,5,main], got Thread[embulk-output-executor-0,5,main]
2017-10-15 19:01:18.898 +0900 [WARN] (embulk-output-executor-1): Owner thread expected Thread[0014:task-0000,5,main], got Thread[embulk-output-executor-1,5,main]
2017-10-15 19:01:18.902 +0900 [WARN] (embulk-output-executor-2): Owner thread expected Thread[0014:task-0000,5,main], got Thread[embulk-output-executor-2,5,main]
2017-10-15 19:01:20.088 +0900 [INFO] (0001:transaction): {done:  1 / 1, running: 0}
org.embulk.exec.PartialExecutionException: java.lang.ArrayIndexOutOfBoundsException: 512
	at org.embulk.exec.BulkLoader$LoaderState.buildPartialExecuteException(BulkLoader.java:375)
	at org.embulk.exec.BulkLoader.doRun(BulkLoader.java:607)
	at org.embulk.exec.BulkLoader.access$000(BulkLoader.java:35)
	at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:391)
	at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:387)
	at org.embulk.spi.Exec.doWith(Exec.java:25)
	at org.embulk.exec.BulkLoader.run(BulkLoader.java:387)
	at org.embulk.EmbulkEmbed.run(EmbulkEmbed.java:180)
	at org.embulk.EmbulkRunner.runInternal(EmbulkRunner.java:337)
	at org.embulk.EmbulkRunner.run(EmbulkRunner.java:174)
	at org.embulk.cli.EmbulkRun.runSubcommand(EmbulkRun.java:467)
	at org.embulk.cli.EmbulkRun.run(EmbulkRun.java:99)
	at org.embulk.cli.Main.main(Main.java:28)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 512
	at org.apache.orc.impl.RunLengthIntegerWriterV2.write(RunLengthIntegerWriterV2.java:725)
	at org.apache.orc.impl.WriterImpl$StringBaseTreeWriter.flushDictionary(WriterImpl.java:1336)
	at org.apache.orc.impl.WriterImpl$StringBaseTreeWriter.createRowIndexEntry(WriterImpl.java:1387)
	at org.apache.orc.impl.WriterImpl$TreeWriter.createRowIndexEntry(WriterImpl.java:761)
	at org.apache.orc.impl.WriterImpl.createRowIndexEntry(WriterImpl.java:2644)
	at org.apache.orc.impl.WriterImpl.flushStripe(WriterImpl.java:2650)
	at org.apache.orc.impl.WriterImpl.checkMemory(WriterImpl.java:278)
	at org.apache.orc.impl.MemoryManagerImpl.notifyWriters(MemoryManagerImpl.java:184)
	at org.apache.orc.impl.MemoryManagerImpl.addedRow(MemoryManagerImpl.java:172)
	at org.apache.orc.impl.WriterImpl.addRowBatch(WriterImpl.java:2823)
	at org.embulk.output.orc.OrcOutputPlugin$OrcTransactionalPageOutput.add(OrcOutputPlugin.java:303)
	at org.embulk.exec.LocalExecutorPlugin$ScatterTransactionalPageOutput$OutputWorker.call(LocalExecutorPlugin.java:394)
	at org.embulk.exec.LocalExecutorPlugin$ScatterTransactionalPageOutput$OutputWorker.call(LocalExecutorPlugin.java:319)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	Suppressed: java.lang.ArrayIndexOutOfBoundsException: 512
		at org.apache.orc.impl.RunLengthIntegerWriterV2.write(RunLengthIntegerWriterV2.java:729)
		... 16 more
	Suppressed: java.lang.ArrayIndexOutOfBoundsException: 512
		... 17 more
	Suppressed: java.lang.ArrayIndexOutOfBoundsException: 512
		at org.apache.orc.impl.RunLengthIntegerWriterV2.determineEncoding(RunLengthIntegerWriterV2.java:438)
		at org.apache.orc.impl.RunLengthIntegerWriterV2.write(RunLengthIntegerWriterV2.java:761)
		... 16 more

Error: java.lang.ArrayIndexOutOfBoundsException: 512

```

**Comment by yuokada** [comment]
```
---
in:
  type: randomj
  rows: 102400  # throw exception when over 4096
 
  threads: 1
  # default_timezone: Asia/Tokyo
  primary_key: myid
  schema:
    - {name: myid,     type: long}
    - {name: named,    type: string}
    - {name: x_flag,   type: boolean}
    - {name: pit_rate, type: double}
    - {name: score,    type: long}
    - {name: time,     type: timestamp, format: '%Y-%m-%d %H:%M:%S'}
    - {name: purchase, type: timestamp, format: '%Y/%m/%d'}

#exec:
#  max_threads: 6         # run at most 8 tasks concurrently
#  min_output_tasks: 2    # disable page scattering

out:
  type: orc
  overwrite:   true
  path_prefix: "/tmp/output"
  buffer_size: 8000
  strip_size:  90000
  compression_kind: ZLIB

```

**Comment by babywhale-sk** [comment]
I also met the same problem when inserting data into oracle with overwrite mode 

09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 2017-11-09 10:10:38.511 +0800 [INFO] (0001:cleanup): > 0.03 seconds
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 2017-11-09 10:10:38.511 +0800 [INFO] (0001:cleanup): SQL: DROP TABLE "TD_USER_ID_223863d00_bl_tmp049"
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 2017-11-09 10:10:38.542 +0800 [INFO] (0001:cleanup): > 0.03 seconds
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - org.embulk.exec.PartialExecutionException: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.exec.BulkLoader$LoaderState.buildPartialExecuteException(org/embulk/exec/BulkLoader.java:373)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.exec.BulkLoader.doRun(org/embulk/exec/BulkLoader.java:591)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.exec.BulkLoader.access$000(org/embulk/exec/BulkLoader.java:33)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.exec.BulkLoader$1.run(org/embulk/exec/BulkLoader.java:389)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.exec.BulkLoader$1.run(org/embulk/exec/BulkLoader.java:385)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.spi.Exec.doWith(org/embulk/spi/Exec.java:25)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.exec.BulkLoader.run(org/embulk/exec/BulkLoader.java:385)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.EmbulkEmbed.run(org/embulk/EmbulkEmbed.java:180)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:497)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.javasupport.JavaMethod.invokeDirectWithExceptionHandling(org/jruby/javasupport/JavaMethod.java:453)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.javasupport.JavaMethod.invokeDirect(org/jruby/javasupport/JavaMethod.java:314)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at RUBY.run(/home/hadoop/.embulk/bin/embulk!/embulk/runner.rb:84)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at RUBY.run(/home/hadoop/.embulk/bin/embulk!/embulk/command/embulk_run.rb:307)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at RUBY.<main>(/home/hadoop/.embulk/bin/embulk!/embulk/command/embulk_main.rb:2)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Ruby.runInterpreter(org/jruby/Ruby.java:850)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Ruby.loadFile(org/jruby/Ruby.java:2976)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.RubyKernel.requireCommon(org/jruby/RubyKernel.java:963)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:956)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.RubyKernel$INVOKER$s$1$0$require19.call(org/jruby/RubyKernel$INVOKER$s$1$0$require19.gen)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at RUBY.(root)(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at home.hadoop.$_dot_embulk.bin.embulk.embulk.command.embulk_bundle.invokeOther66:require(home/hadoop/$_dot_embulk/bin/embulk/embulk/command/file:/home/hadoop/.embulk/bin/embulk!/embulk/command/embulk_bundle.rb:51)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at home.hadoop.$_dot_embulk.bin.embulk.embulk.command.embulk_bundle.<main>(file:/home/hadoop/.embulk/bin/embulk!/embulk/command/embulk_bundle.rb:51)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at java.lang.invoke.MethodHandle.invokeWithArguments(java/lang/invoke/MethodHandle.java:625)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Ruby.runScript(org/jruby/Ruby.java:834)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Ruby.runNormally(org/jruby/Ruby.java:749)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Ruby.runNormally(org/jruby/Ruby.java:767)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Ruby.runFromMain(org/jruby/Ruby.java:580)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Main.doRunFromMain(org/jruby/Main.java:425)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Main.internalRun(org/jruby/Main.java:313)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Main.run(org/jruby/Main.java:242)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.jruby.Main.main(org/jruby/Main.java:204)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.cli.Main.main(org/embulk/cli/Main.java:23)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - Caused by: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at java.lang.System.arraycopy(Native Method)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at oracle.jdbc.driver.OraclePreparedStatement.executeBatch(OraclePreparedStatement.java:11107)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at oracle.jdbc.driver.OracleStatementWrapper.executeBatch(OracleStatementWrapper.java:244)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.output.jdbc.StandardBatchInsert.flush(StandardBatchInsert.java:75)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.output.jdbc.AbstractJdbcOutputPlugin$PluginPageOutput$3.run(AbstractJdbcOutputPlugin.java:974)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.output.jdbc.AbstractJdbcOutputPlugin$RetryableSQLExecution.call(AbstractJdbcOutputPlugin.java:1088)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.output.jdbc.AbstractJdbcOutputPlugin$RetryableSQLExecution.call(AbstractJdbcOutputPlugin.java:1076)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.spi.util.RetryExecutor.run(RetryExecutor.java:100)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.spi.util.RetryExecutor.runInterruptible(RetryExecutor.java:77)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.output.jdbc.AbstractJdbcOutputPlugin.withRetry(AbstractJdbcOutputPlugin.java:1060)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.output.jdbc.AbstractJdbcOutputPlugin.withRetry(AbstractJdbcOutputPlugin.java:1053)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.output.jdbc.AbstractJdbcOutputPlugin$PluginPageOutput.add(AbstractJdbcOutputPlugin.java:970)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.exec.LocalExecutorPlugin$ScatterTransactionalPageOutput$OutputWorker.call(LocalExecutorPlugin.java:394)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at org.embulk.exec.LocalExecutorPlugin$ScatterTransactionalPageOutput$OutputWorker.call(LocalExecutorPlugin.java:319)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	at java.lang.Thread.run(Thread.java:745)
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 	Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 		Suppressed: java.lang.ArrayIndexOutOfBoundsException
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 			... 18 more
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - 
09-11-2017 10:10:38 CST 3fdef228-8623-44c8-82ef-a85fcd97e170 INFO - Error: java.lang.ArrayIndexOutOfBoundsException

**Comment by babywhale-sk** [comment]
anyidea for quick workaroud this ?

**Comment by yuokada** [comment]
@babywhale-sk  I have no idea for quick workaround now.

Which version do you use?

**Comment by yuokada** [comment]
see: [[ORC-361] org.apache.orc.impl.MemoryManagerImpl : Owner thread expected Thread[main,5,main], got Thread[pool-15-thread-1,5,main] - ASF JIRA](https://issues.apache.org/jira/browse/ORC-361 "[ORC-361] org.apache.orc.impl.MemoryManagerImpl : Owner thread expected Thread[main,5,main], got Thread[pool-15-thread-1,5,main] - ASF JIRA")

**Comment by kyontan** [comment]
I have same problem, using `embulk-output-orc-0.3.2`

**Comment by yuokada** [comment]
@kyontan Sorry, this issue isn't fix yet.

Workaround:
- set `min_output_tasks` to 1 
    ```
    exec:
      max_threads: 2         # run at most 8 tasks concurrently
      min_output_tasks: 1    # disable page scattering
    ```
- set `buffer_size` to more huge value

**Comment by yzp0n** [comment]
> see: [[ORC-361] org.apache.orc.impl.MemoryManagerImpl : Owner thread expected Thread[main,5,main], got Thread[pool-15-thread-1,5,main] - ASF JIRA](https://issues.apache.org/jira/browse/ORC-361)

There's a chance this is related to #16

**Comment by yuokada** [comment]
It be fixed version 0.3.4
https://rubygems.org/gems/embulk-output-orc/versions/0.3.4
